### ISSUE TITLE
Best pattern with this code: should I move this global var in my func?

### ISSUE THREAD

**Issue by frederikhors** [leighmcculloch/gochecknoglobals#19]
Using this code:

```go
var namePatterns = []map[string]string{
	{`[^\pL-\s']`: ""}, // cut off everything except [alpha, hyphen, whitespace, apostrophe]
	{`\s{2,}`: " "},    // trim more than two whitespaces to one
	{`-{2,}`: "-"},     // trim more than two hyphens to one
	{`'{2,}`: "'"},     // trim more than two apostrophes to one
	{`( )*-( )*`: "-"}, // trim enclosing whitespaces around hyphen
}

var nameRegex = regexp.MustCompile(`[\p{L}]([\p{L}|[:space:]\-']*[\p{L}])*`)

func onlyOne(s string) string {
	for _, v := range namePatterns {
		for f, r := range v {
			s = regexp.MustCompile(f).ReplaceAllLiteralString(s, r)
		}
	}

	return s
}
```

`namePatterns` is reported as a global variable which is correct, but what is the best pattern here?

Should I move it in my `onlyOne` func?

Isn't it slower doing so?

**Comment by leighmcculloch** [comment]
Unfortunately there's not a great story for vars such as these. Ideally Go would provide a way to define this as a constant, or as an immutable var which would eliminate side effects, but those features don't exist. So as a linter we can choose to add configurability and allow ignoring cases like these, or approach is purely and say no to them. Gochecknoglobals does the latter to reduce knobs.

**Comment by frederikhors** [comment]
So I should use `//nolint`, right?

**Comment by frederikhors** [comment]
ping @leighmcculloch

**Comment by leighmcculloch** [comment]
Good question. Using a nolint comment if you're using golangci-lint would be one way to get past it. Whether that is the best practice or not I cannot say. I try to avoid nolints, but sometimes there is no other choice.
